vyper/context/types/function.py

```diff
@@ -167,8 +180,10 @@ def from_FunctionDef(
                             "Visibility must be public or private, not both", node
                         )
                     kwargs["is_public"] = bool(decorator.id == "public")
-                elif decorator.id in ("constant", "payable"):
-                    kwargs[f"is_{decorator.id}"] = True
+                elif decorator.id == "payable":
+                    kwargs["is_payable"] = True
+                elif decorator.id == "view":
+                    kwargs["is_constant"] = True
                 else:
                     raise FunctionDeclarationException(
```

Comment:
It might make sense to temporarily add a special exception here for when the decorator is `@constant`, to let people know about the change.

Reply:
Fixed in 0c5023b